### ISSUE TITLE
Server-sent events

### ISSUE THREAD

**Issue by zxey** [uNetworking/uWebSockets#919]
Is there a way to use server-sent events with uWebSockets? By sse I mean [this](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).
I tried to do this by streaming data from server, similarly to how you use AsyncFileStreamer, but it does not work.
There's also mdn example showing that it should not be that hard implementing sse [here](https://github.com/mdn/dom-examples/blob/master/server-sent-events/sse.php)
Could you point me to the right direction?

**Comment by zxey** [comment]
I tried something along these lines:
```
uWS::App()
    .get("/sse", [](auto *res, auto *req) {
        res->onAborted([res]() {
            std::cerr << "aborted" << std::endl;
        });

        res->onWritable([res](int offset) {
            res->write("event: test");
            return false;
        });

        res->writeStatus("200 OK");
        res->writeHeader("Content-Type", "text/event-stream");
        res->writeHeader("Cache-Control", "no-cache");
    });
```

**Comment by alexhultman** [comment]
You are almost there but you need to read the docs.

**Comment by ngzhian** [comment]
@zxey in case you're still figuring this out, I wanted to use SSE to have live reload for blogging. https://github.com/ngzhian/blog/commit/e76e9643653ecb657ebbcca2973c96e5b6c7cc37#diff-b388e57e18c6815f3bdd3a6412184385 is what I ended up with.

I used uWebSockets to listen on 9001, when I do `curl localhost:9001` it sends a message to a connected session (by keeping the HttpResponse in a global object). The browser first establishes the connection by creating a `new EventSource('http://localhost:9001/eventsource')`, see the same commit. So far it has worked okay for me.
